scylla-cdc-base/src/main/java/com/scylladb/cdc/model/worker/Worker.java

```diff
@@ -89,9 +98,29 @@ private static TaskState getInitialStateForStreams(Map<TaskId, SortedSet<StreamI
      * This includes fetching saved state of each task or creating a new initial
      * state for tasks that haven't run successfully before.
      */
-    private TaskActionsQueue queueFirstActionForEachTask(Map<TaskId, SortedSet<StreamId>> groupedStreams) throws ExecutionException, InterruptedException {
-        return new TaskActionsQueue(createTasksWithState(groupedStreams)
-                .map(task -> TaskAction.createFirstAction(workerConfiguration, task)).collect(Collectors.toSet()));
+    private Collection<TaskAction> queueFirstActionForEachTask(Map<TaskId, SortedSet<StreamId>> groupedStreams)
+            throws ExecutionException, InterruptedException {
+        return createTasksWithState(groupedStreams).map(task -> TaskAction.createFirstAction(workerConfiguration, task))
+                .collect(Collectors.toSet());
+    }
+
+    private boolean shouldStop() {
+        return shouldStop;
+    }
+
+    private ScheduledExecutorService getExecutorService() {
+        return workerConfiguration.getExecutorService();
+    }
+
+    private Callable<Object> makeCallable(TaskAction a) {
+        return () -> a.run().handleAsync((na, ex) -> {
+            if (ex != null) {
+                logger.atSevere().withCause(ex).log("Unhandled exception in Worker.");
```

Comment:
Is this ok to just stop doing anything for the task? We won't be able to retry, will we? @avelanarius please comment.

Reply:
The task themselves should handle exceptions (and issue retries). I believe all of our task do that (in their `run()` methods) - no exception should "bubble up" to this `handleAsync`.

This logging is for a case when for some reason, a task would not catch exception - without this logging it would fail silently (at least that was a case in previous queue-based implementation).

Reply:
ok. thanks.

Reply:
Very much so. That is why the "severe".